src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5855,6 +5867,208 @@ private BoundExpression BindTypeParameterCreationExpression(ObjectCreationExpres
+                targetType);
+        }
+
+        private BoundExpression BindCollectionLiteralElement(
+            CollectionElementSyntax syntax,
+            bool hasEnumerableInitializerType,
+            Binder collectionInitializerAddMethodBinder,
+            BindingDiagnosticBag diagnostics,
+            BoundObjectOrCollectionValuePlaceholder implicitReceiver)
+        {
+            switch (syntax)
+            {
+                case ExpressionElementSyntax expressionElementSyntax:
+                    {
+                        var elementSyntax = expressionElementSyntax.Expression;
+                        var element = BindValue(elementSyntax, diagnostics, BindValueKind.RValue);
+
+                        var elementType = element.Type;
+                        if (elementType.IsKeyValuePair())
+                        {
+                            // PROTOTYPE: Create the BoundPropertyAccess expressions for key and value in lowering.
+                            var key = getPropertyValue(expressionElementSyntax, element, "Key", diagnostics);
+                            var value = getPropertyValue(expressionElementSyntax, element, "Value", diagnostics);
+                            return bindDictionaryElementInitializer(syntax, implicitReceiver, key, value, diagnostics);
+                        }
+                        else
+                        {
+                            BoundExpression result = BindCollectionInitializerElementAddMethod(
+                                elementSyntax,
+                                ImmutableArray.Create(element),
+                                hasEnumerableInitializerType,
+                                collectionInitializerAddMethodBinder,
+                                diagnostics,
+                                implicitReceiver);
+                            result.WasCompilerGenerated = true;
+                            return result;
+                        }
+                    }
+
+                case DictionaryElementSyntax dictionaryElementSyntax:
+                    {
+                        var key = BindValue(dictionaryElementSyntax.KeyExpression, diagnostics, BindValueKind.RValue);
+                        var value = BindValue(dictionaryElementSyntax.ValueExpression, diagnostics, BindValueKind.RValue);
+                        return bindDictionaryElementInitializer(syntax, implicitReceiver, key, value, diagnostics);
+                    }
+
+                case SpreadElementSyntax spreadElementSyntax:
+                    {
+                        var elementSyntax = spreadElementSyntax.Expression;
+                        var element = BindValue(elementSyntax, diagnostics, BindValueKind.RValue);
+
+                        // PROTOTYPE: What is the required type of the spread element? Is it either IEnumerable<T>
+                        // or a type that implements the "IEnumerable<T> pattern"? What about IEnumerable?
```

Comment:
the spec covers this.  It's implied by the translation of the spread element, which is effectively:

```
foreach (TargetElementType v in __spreadElement)
    __destination[index++] = v;
```

So this would work without IEnumerable as long as the source is iterable in some fashion.